manifests/record.pp

```diff
@@ -0,0 +1,38 @@
+# Define new DNS record
+define dns::record (
+  $zone     = undef,
+  $label    = undef,
+  $target   = undef,
+  $type     = undef,
+  $priority = undef,
+  $comment  = ' ',
+)
+{
+
+  $dnstype = upcase($type)
+
+  if ($label == undef) or ($zone == undef) {
+    $domain_array = split($title, '[.]' )
+    $_label = $domain_array[0]
+    $_zone = join(delete_at($domain_array, 0), '.')
+  } else {
+    $_label = $label
+    $_zone = $zone
+  }
+  
+  if $dnstype =~ /(CNAME|SRV|TXT|A|PTR)/ {
+    $line = "${_label} IN ${dnstype} ${target}  ; ${comment}"
+  }
+  elsif $dnstype == 'MX' {
+    $line = "${_label} IN ${dnstype} ${priority} ${target}  ; ${comment}"
+  }
+  else {
+    warning('Incorrect DNS record type specified')
+  }
+
+  concat_fragment { "dns-static-${_zone}+${title}.dnsstatic":
```

Comment:
I'd include ```02content``` in there so you're sure the header is always first (00 is a valid DNS label): ```dns-static-${_zone}+02content-${title}.dnsstatic```.